### PR TITLE
Support nightly RPM builds

### LINKED ISSUE #1366: Build nightly RPMs
To support nightlies, there are changes necessary to `.gitlab-ci.yml` to support nightly builds and produce sane versions.  Then also need to add a pipeline schedule to OSU Gitlab.

@johrstrom Do we want nightlies installed on OSC dev OnDemand servers?

### PR DESCRIPTION
**Pull request by treydock**
Fixes #1366 

The version of nightly RPM where latest OnDemand tag is `v2.0.15` would be `v2.0.15-99.20210825.nightly`.  So each days build will be seen as an upgrade by YUM/DNF.  

I think one way to test this is not merge, but leave in this branch and setup gitlab CI pipeline schedule to build off this branch to test with.

I already bootstrapped the nightly repos using ondemand-packaging scripts.

### CODE REVIEW

**Comment by treydock** [comment]
@johrstrom The manual builds of nightlies now work. I've enabled the schedule to run daily at 7AM so will see how that goes tomorrow then this will be ready.

**Comment by treydock** [comment]
This is new version I think I'm sticking with:

```
ondemand-2.0.20210902-1.b037d8b.nightly.el7.x86_64.rpm
```

The `2.0` part is necessary to pull in correct runtime dependencies at build time.  I also wanted git SHA in release to make it clear which commit was used for build if someone wanted to know at a glance.

**Comment by treydock** [comment]
@johrstrom I think this is ready. I also removed CI repo RPM upload.  The nightlies last night were successful and I just ran a manual build with the new version and that was successful.  The Gitlab schedule is set for every morning at 7AM.  Once this is merged I can update the schedule to target the `master` branch

**Comment by treydock** [comment]
Actually hold off merging. I looked at code and shouldn't be too hard to update Gitlab CI to use Rake tasks instead of raw bash to build the RPMs.  I will test that out in this PR for nightlies since that's easy to test and trigger.

**Comment by treydock** [comment]
New version that ensures two nightlies (ie manual build) on same day properly upgrade:

```
ondemand-2.0.20210902-107602.049d89f.nightly.el7.x86_64.rpm
```

Also used Rake task for building RPMs with Gitlab CI.

**Comment by treydock** [comment]
@johrstrom I think this is ready if looks good to you.  The Puppet module for OnDemand ran some acceptance tests to validate installing `ondemand` RPM from latest repo then upgrading to nightly build.

**Review by johrstrom — APPROVED** [review]
Works for me. I think the whole 'how to determine the version' thing could use some cleanup, but we can defer that.

**Pull request merged**
